src/libcore/sync/atomic.rs

```diff
@@ -1001,395 +996,455 @@ macro_rules! atomic_int {
         unsafe impl Sync for $atomic_type {}
 
         impl $atomic_type {
-            /// Creates a new atomic integer.
-            ///
-            /// # Examples
-            ///
-            /// ```
-            /// use std::sync::atomic::AtomicIsize;
-            ///
-            /// let atomic_forty_two  = AtomicIsize::new(42);
-            /// ```
-            #[inline]
-            #[$stable]
-            pub const fn new(v: $int_type) -> Self {
-                $atomic_type {v: UnsafeCell::new(v)}
+            doc_comment! {
+                concat!("Creates a new atomic integer.
+
+# Examples
+
+```
+#![feature(integer_atomics)]
```

Comment:
Will this feature gate appear on every atomic's docs? It may be misleading to have this on (for example) `AtomicUsize` which doesn't require it.

Reply:
Indeed!